hlv/hlv.go

```diff
@@ -82,22 +82,24 @@ type currentVersion struct {
 // If cas > ver, then the document has been mutated since replication.
 // - If there is no mv, the input cv will be rolled into pv
 // - If there is an mv, the mv will be rolled into pv
-func NewHLV(source DocumentSourceId, cas uint64, src DocumentSourceId, ver uint64, pv, mv VersionsMap) (*HLV, error) {
+func NewHLV(source DocumentSourceId, cas uint64, cvCas uint64, src DocumentSourceId, ver uint64, pv, mv VersionsMap) (*HLV, error) {
 	hlv := HLV{
 		cvCAS: cas,
 	}
-	if cas < ver {
-		// ver is initially the same as cas. Cas may increase later. So ver should never be greater than cas
-		return nil, fmt.Errorf("cas < ver, cas=%v,ver=%v,src=%v,pv=%s,mv=%s", cas, ver, src, pv, mv)
-	} else if cas == ver {
+	if cas < cvCas {
+		// cvCas is initially the same as cas. Cas may increase later. So cvCas should never be greater than cas
```

Comment:
Let's rephrase this comment to communicate the right idea for cvCAS
When cvCAS is the same as CAS, this means X
When they are different, this means Y